src/base.ts

```diff
@@ -79,38 +75,68 @@ export const resolveServiceUrl = (service: Service): string => {
   throw new Error(`Missing endpoint configuration for "${service}" service`)
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+const getAuthorizationHeader = (accessToken = null) => {
+  if (!accessToken) {
+    throw new Error(
+      "You tried to make an authenticated requests without providing an access token!\n Please pass a valid token as a request option."
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return { Authorization: `Bearer ${accessToken}` }
+}
+
+export interface RequestOptions {
+  isAuthorizedRequest?: boolean
+  accessToken?: string
+  method?: "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+  headers?: HeadersInit
 }
```

Comment:
In my PR I created two interfaces, one private for fetch wrappers and one public for the calls: https://github.com/carforyou/carforyou-api-client-pkg/pull/179/files#diff-440b55a9ff9e0486020fc49197f3278d14446306fc0c5d5107b48139c20bfda6R41-R51

This might be a good idea here too. Otherwise you open yourself to a situation where a consumer can call:

`getListing({ id: 123, method: "POST" })` or `getDealerListings({ dealerId: 123, isAuthorizedRequest: false })`

Reply:
I thought about that too and went with a unified interface to make it simpler to work with as you don't need to determine between internal and external calls and it provides the max amount of flexibility. Yes someone could fuck up the request like that but I expect some logical sense when consuming it to not switch pre configured settings. There might be cases where it gets handy to be have to override them for test cases or so if you are working with mock responses and don't need to pass tokens so you could tweak the behavior to not throw for those cases...

I don't mind switching in you PR, do you think we need to do the same "twice" or will it be fine if we just split the interfaces in your PR?